js/src/Axis.ts

```diff
@@ -57,7 +59,7 @@ export class Axis extends WidgetView {
         this.listenTo(this.model, "change:scale", (model, value) => {
             this.update_scale(model.previous("scale"), value);
             // TODO: rescale_axis does too many things. Decompose
-            this.axis.scale(this.axis_scale.scale); // TODO: this is in redraw_axisline
+            this.axis.scale(this.axis_scale.scale as d3.AxisScale<d3.AxisDomain>); // TODO: this is in redraw_axisline
```

Comment:
Instead of doing this, I guess we want to define `axis_scale` as being a `Scale` widget view, then define `scale` as being a `d3.AxisScale<d3.AxisDomain>` in the `Scale` widget implementation. Then this would work without the need of casting.

Reply:
My comment does not make any sense. Sorry for that. Not sure how we could get rid of this cast!